internal/receiver/databricksreceiver/databricks_client.go

```diff
@@ -14,70 +14,74 @@
 
 package databricksreceiver
 
-import "fmt"
+import (
+	"encoding/json"
+	"fmt"
+)
 
-// databricksClientInterface is extracted from databricksClient for swapping out in unit tests
-type databricksClientInterface interface {
-	jobs() (out []job, err error)
-	activeJobRuns() (out []jobRun, err error)
-	completedJobRuns(jobID int, time int64) (out []jobRun, err error)
+// databricksClient wraps a databricksRawClient implementation and unmarshals json byte
+// arrays to the types defined in json_types.go. Its method signatures mirror
+// those of the rawClient.
+type databricksClient struct {
+	rawClient databricksRawClient
 }
 
-// databricksClient handles pagination (responses specify hasMore=true/false) and
-// combines the returned objects into one array.
-type databricksClient struct {
-	unmarshaller unmarshaller
-	limit        int
+func (c databricksClient) jobsList(limit int, offset int) (jobsList, error) {
+	bytes, err := c.rawClient.jobsList(limit, offset)
+	out := jobsList{}
+	if err != nil {
+		return out, fmt.Errorf("databricksClient.jobsList(): %w", err)
+	}
+	err = json.Unmarshal(bytes, &out)
+	return out, err
+}
+
+func (c databricksClient) activeJobRuns(limit int, offset int) (jobRuns, error) {
+	bytes, err := c.rawClient.activeJobRuns(limit, offset)
+	out := jobRuns{}
+	if err != nil {
+		return out, fmt.Errorf("databricksClient.activeJobRuns(): %w", err)
+	}
+	err = json.Unmarshal(bytes, &out)
+	return out, err
 }
 
-func newDatabricksClient(api apiClientInterface, limit int) databricksClient {
-	return databricksClient{
-		unmarshaller: unmarshaller{api: api},
-		limit:        limit,
+func (c databricksClient) completedJobRuns(jobID int, limit int, offset int) (jobRuns, error) {
+	bytes, err := c.rawClient.completedJobRuns(jobID, limit, offset)
+	out := jobRuns{}
+	if err != nil {
+		return out, fmt.Errorf("databricksClient.completedJobRuns(): %w", err)
 	}
+	err = json.Unmarshal(bytes, &out)
+	return out, err
 }
 
-func (c databricksClient) jobs() (out []job, err error) {
-	hasMore := true
-	for i := 0; hasMore; i++ {
-		resp, err := c.unmarshaller.jobsList(c.limit, c.limit*i)
-		if err != nil {
-			return nil, fmt.Errorf("databricksClient.jobs(): %w", err)
-		}
-		out = append(out, resp.Jobs...)
-		hasMore = resp.HasMore
+func (c databricksClient) clusterList() (clusterList, error) {
```

Comment:
to mirror the wrapped method?

```suggestion
func (c databricksClient) clustersList() (clusterList, error) {
```

Reply:
Fixed.